save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt

```diff
@@ -103,15 +108,37 @@ class SaveAgent(private val config: AgentConfiguration,
         val code = runSave(emptyList())
         when (code) {
             0 -> {
-                // todo: read data from files here
+                val executionLogs = ExecutionLogs(id, readFile("logs.txt"))
                 sendExecutionData(ExecutionData(emptyList()))
+                sendLogs(executionLogs)
                 state.value = AgentState.FINISHED
             }
             else -> state.value = AgentState.CLI_FAILED
         }
     }
 
-    private fun runSave(cliArgs: List<String>) = platform.posix.system("sleep 5")  // fixme: actually run save CLI here
+    private fun readFile(filePath: String): List<String> {
+        try {
+            val path = filePath.toPath()
+            return FileSystem.SYSTEM.read(path) {
+                generateSequence { readUtf8Line() }.toList()
+            }
+        } catch (e: FileNotFoundException) {
+            println("Not able to find file in the following path: $filePath")
+            exitProcess(1)
```

Comment:
This case means that CLI hasn't written any logs; if this happens, then there are some errors. We shouldn't stop the process silently, but rather change status to `CLI_FAILED` and wait for the next command from orchestrator